python_tests/test_cached_extra_study_property.py

```diff
@@ -78,6 +78,37 @@ def test_different_distributions(self) -> None:
         cached_extra_study_property = _CachedExtraStudyProperty()
         cached_extra_study_property.update(trials)
 
+        self.assertEqual(len(cached_extra_study_property.intersection_search_space), 2)
+        self.assertEqual(len(cached_extra_study_property.union_search_space), 2)
```

Comment:
Could you add an assert statement to check whether x0's distribution in `cached_extra_study_property.union_search_space` is `FloatDistrubution(low=0, high=10)`?